src/core/unif.ml

```diff
@@ -68,24 +67,30 @@ let instantiation : ctxt -> meta -> term array -> term ->
     match nl_distinct_vars ctx ts with
     | None -> None
     | Some(vs, map) ->
+        (*if !log_enabled then
+          log_unif "variables: %a" (Array.pp pp_var " ") vs;*)
         let u = Eval.simplify (sym_to_var map u) in
-        if !log_enabled then log_unif "variables: %a; value: %a"
-                               (Array.pp pp_var " ") vs pp_term u;
         Some (Bindlib.bind_mvar vs (lift u))
   else None
 
 (** Checking type or not during meta instanciation. *)
 let do_type_check = Stdlib.ref true
 
-(** [instantiate ctx m ts u] check whether, in a problem [m[ts]=u], [m] can be
-    instantiated and, if so, instantiate it. *)
-let instantiate : ctxt -> meta -> term array -> term -> constr list -> bool =
-  fun ctx m ts u initial ->
+(** Initial set of constraints. *)
+let initial : constr list Stdlib.ref = Stdlib.ref []
```

Comment:
Do you think that you could provide an interface for this module? Because I guess that this reference should not be used outside of `unif.ml` right?

Reply:
Right. The same applies to do_type_check. But I am not sure it is worth doing an interface, at least for now.